### PR TITLE
boards: stm32: Add nucleo_h533re support

### PR DESCRIPTION
**Pull request by marwaiehm-st**
Add DTS and SoC support for STM32h533Xe familly
Add board support for STM32H533RE

### CODE REVIEW

File: dts/arm/st/h5/stm32h533Xe.dtsi [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	sram3: memory@20050000 {
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x20050000 DT_SIZE_K(64)>;
+		zephyr,memory-region = "SRAM3";
+	};
+
+	soc {
+		flash-controller@40022000 {
+			flash0: flash@8000000 {
+				reg = <0x08000000 DT_SIZE_K(215)>;
```
Comment (erwango) [inline dts/arm/st/h5/stm32h533Xe.dtsi:31]:
```suggestion
				reg = <0x08000000 DT_SIZE_K(512)>;
```
?

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: boards/st/nucleo_h533re/nucleo_h533re.dts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+#include "arduino_r3_connector.dtsi"
+#include "st_morpho_connector.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "STMicroelectronics STM32H533RE-NUCLEO board";
+	compatible = "st,stm32h533re-nucleo";
+
+	#address-cells = <1>;
+	#size-cells = <1>;
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/nucleo_h533re.dts:19]: Is this needed?

File: boards/st/nucleo_h533re/nucleo_h533re.dts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	chosen {
+		zephyr,console = &usart2;
+		zephyr,shell-uart = &usart2;
+		zephyr,sram = &sram1;
+		zephyr,flash = &flash0;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led_2: led_42{
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/nucleo_h533re.dts:30]: nit
```suggestion
		green_led_2: led_42 {
```

File: boards/st/nucleo_h533re/nucleo_h533re.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+arch: arm
+toolchain:
+  - zephyr
+ram: 272
+flash: 512
+supported:
+  - gpio
+  - watchdog
+  - pwm
+  - dma
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/nucleo_h533re.yaml:16]:
```
- rtc
```

File: boards/st/nucleo_h533re/nucleo_h533re.dts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+&clk_lse {
+	status = "okay";
+};
+
+&rtc {
+	clocks = <&rcc STM32_CLOCK_BUS_APB3 0x00200000>,
+		 <&rcc STM32_SRC_LSE RTC_SEL(1)>;
+	status = "okay";
+};
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/nucleo_h533re.dts:129]: Could you add an overlay for this board in `tests/drivers/rtc/rtc_api/boards` as well?

File: boards/st/nucleo_h533re/nucleo_h533re.dts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+&gpdma1 {
+	status = "okay";
+};
+
+&gpdma2 {
+	status = "okay";
+};
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/nucleo_h533re.dts:119]: Do these need to be enabled?

File: boards/st/nucleo_h533re/nucleo_h533re.yaml (hunk 4 above)
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/nucleo_h533re.yaml:16]: have you tested `dma`? also for this tag to be included I think an overlay should be added to `tests/drivers/dma/loop_transfer/boards`

File: boards/st/nucleo_h533re/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,317 @@
+.. _nucleo_h533re_board:
+
+ST Nucleo H533re
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/doc/index.rst:3]: for consistency across doc
```suggestion
ST Nucleo H533RE
```

File: boards/st/nucleo_h533re/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+   - VBUSC from the USB user connector (5V, 0.5A)
+   - 3V3_EXT supplied via a pin header CN6 pin 4 or CN7 pin 16 (3.3V, 1.3A)
+
+- On-board ST-LINK/V3EC debugger/programmer
+
+  - mass storage
+  - Virtual COM port
+  - debug port
+
+- One user LED shared with ARDUINO® Uno V3
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/doc/index.rst:38]:
```suggestion
- One user LED shared with ARDUINO |reg| Uno V3
```

File: boards/st/nucleo_h533re/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+The Zephyr nucleo_h533re board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| ADC       | on-chip    | ADC Controller                      |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| DAC       | on-chip    | DAC Controller                      |
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/doc/index.rst:158]: Not enabled in DTS

File: boards/st/nucleo_h533re/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | PWM                                 |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+| RTC       | on-chip    | Real Time Clock                     |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi bus                             |
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/doc/index.rst:172]: Not enabled in DTS

File: boards/st/nucleo_h533re/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c bus                             |
```
Comment (ajarmouni-st) [inline boards/st/nucleo_h533re/doc/index.rst:174]: Not enabled in DTS

**Review by ajarmouni-st — CHANGES_REQUESTED** [review]

File: boards/st/nucleo_h533re/nucleo_h533re.yaml (hunk 4 above)
Comment (marwaiehm-st) [reply boards/st/nucleo_h533re/nucleo_h533re.yaml:16]: No i didn't tested the dma, i think should be removed from this PR and treated on sperated one.

File: boards/st/nucleo_h533re/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+You should see the following message on the console:
+
+.. code-block:: console
+
+   Hello World! nucleo_h533re
+
+Debugging
+=========
+
+You can debug an application in the usual way.  Here is an example for the
```
[the remark this answers is not in the document]
Comment (marwaiehm-st) [reply boards/st/nucleo_h533re/doc/index.rst]: No, PyOCD is supported

**Review by ajarmouni-st — DISMISSED** [review]

**Review by gautierg-st — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]

File: boards/st/nucleo_h533re/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+
+- On-board ST-LINK/V3EC debugger/programmer
+
+  - mass storage
+  - Virtual COM port
+  - debug port
+
+- One user LED shared with ARDUINO |reg| Uno V3
+- Two push-buttons: USER and RESET
+- 32.789 kHz crystal oscillator
```
Comment (FRASTM) [inline boards/st/nucleo_h533re/doc/index.rst]:
```suggestion
- 32.768 kHz crystal oscillator
```

File: boards/st/nucleo_h533re/doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,306 @@
...
+- SPI1 SCK/MISO/MOSI/NSS: PA5/PA6/PA7/PC9
+- UART2 TX/RX : PA2/PA3 (VCP)
+- USER_PB : PC13
+
+System Clock
+------------
+
+Nucleo H533RE System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at
+240MHz, driven by 8MHz external clock provided from the STLINK-V3EC.
```
Comment (FRASTM) [inline boards/st/nucleo_h533re/doc/index.rst:219]: There is a 24MHz (X3) for the HSE external clock on the nucleo_h533 board
--> HSE is not STLINK sourced

File: boards/st/nucleo_h533re/nucleo_h533re.dts [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+
+&clk_hsi {
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&clk_hsi>;
+	clock-frequency = <DT_FREQ_M(64)>;
+	ahb-prescaler = <1>;
```
Comment (FRASTM) [inline boards/st/nucleo_h533re/nucleo_h533re.dts:71]: we Could also increase the sysclock with the  PLL

File: soc/st/stm32/stm32h5x/Kconfig.defconfig.stm32h533xx [hunk 16]
```diff
@@ -0,0 +1,11 @@
+# STMicroelectronics STM32H503XX MCU
+
+# Copyright (c) 2024 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32H533XX
+
+config NUM_IRQS
+	default 134
```
Comment (FRASTM) [inline soc/st/stm32/stm32h5x/Kconfig.defconfig.stm32h533xx:9]: can you check, RM0481 gives only positions 0 to  132 for the  STM32H523/533xx vector table

**Review by FRASTM — COMMENTED** [review]
LGTM

**Review by ajarmouni-st — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**